packages/SqueakMaps-GeoServices.package/SMANominatimGeocoder.class/instance/locate..st

```diff
@@ -1,23 +1,11 @@
 searching
 locate: aQueryString
 
-	| searches queryUrlString json coordinates | 
+	| queryUrlString json | 
 
-	searches := OrderedCollection new. 
 	queryUrlString := self queryUrlFor: aQueryString.
 	json := self contentSource getJsonFrom: queryUrlString.
-	
-	json do: [:result |
-		coordinates := SMAGeoCoordinate newLong: (result at: 'lon') asNumber Lat: (result at: 'lat') asNumber.
-		
-		searches add: (SMALocation new
-			label: (result at: 'display_name' ifAbsent: [aQueryString]);
-			coordinates: coordinates;
-			boundingBox: (self getBoundingBoxFrom: result);
-			yourself)].
-	
-	searches ifEmpty: [
+	^ (json collect: [:result | self createLocationFrom: result with: aQueryString])
+		ifEmpty: [
 		(SMALocationNameNotFoundError new
-			requestedName: aQueryString) signal].
-
-	^ searches
\ No newline at end of file
+			requestedName: aQueryString) signal]
```

Comment:
```
json do: [:result |
		coordinates := SMAGeoCoordinate newLong: (result at: 'lon') asNumber Lat: (result at: 'lat') asNumber.

		searches add: (SMALocation new
			label: (result at: 'display_name' ifAbsent: [aQueryString]);
			coordinates: coordinates;
			boundingBox: (self getBoundingBoxFrom: result);
			yourself)].

	searches ifEmpty: [
	^ (json collect: [:result | self createLocationFrom: result with: aQueryString])
		ifEmpty: [
		(SMALocationNameNotFoundError new
			requestedName: aQueryString) signal].

	^ searches 
			requestedName: aQueryString) signal] 
```
Hier muss ich sagen, fänd ich es besser, wenn das collect erst mal separat in `searches` gespeichert wird und dann separat auf searches `ifEmpty` aufgerufen und der Fehler geworfen wird. Das finde ich deutlich verständlicher als das aus dem Json zu collecten und direkt auf dem Ergebnis `ifEmpty` aufzurufen. Dass ihr jedoch die Methode `createLocationFrom` eingefügt und das `json do:´ zu `json collect:` geändert habt, finde ich gut 👍

Reply:
Gefällts dir so wies jetzt ist?

Reply:
Ja sieht gut aus, hab mir die neuen Änderungen angeguckt und ihr könnt es meinetwegen mergen, ich würde es approven 👍🏼